subcommand/server-acl-init/command.go

```diff
@@ -117,95 +131,146 @@ func (c *Command) Run(args []string) int {
-		var httpPort int32
-		for _, p := range pod.Spec.Containers[0].Ports {
-			if p.Name == "http" {
-				httpPort = p.ContainerPort
-			}
-		}
-		if httpPort == 0 {
-			c.UI.Error(fmt.Sprintf("pod %s has no port labeled 'http'", pod.Name))
+	// The ClientSet might already be set if we're in a test.
+	if c.clientset == nil {
+		if err := c.configureKubeClient(); err != nil {
+			logger.Error(err.Error())
 			return 1
 		}
-		addr := fmt.Sprintf("%s:%d", pod.Status.PodIP, httpPort)
-		serverPodAddrs = append(serverPodAddrs, addr)
 	}
-	logger.Info(fmt.Sprintf("Found %d Consul servers", len(serverPodAddrs)),
-		"addrs", strings.Join(serverPodAddrs, ","))
 
-	// Pick the first pod to connect to for bootstrapping and set up connection.
-	firstServerAddr := serverPodAddrs[0]
-	consulClient, err := api.NewClient(&api.Config{
-		Address: firstServerAddr,
-		Scheme:  "http",
-	})
+	// Wait if there's a rollout of servers.
+	ssName := c.flagReleaseName + "-consul-server"
+	err = c.untilSucceeds(fmt.Sprintf("waiting for rollout of statefulset %s", ssName), func() error {
+		ss, err := c.clientset.AppsV1().StatefulSets(c.flagNamespace).Get(ssName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		if ss.Status.CurrentRevision == ss.Status.UpdateRevision {
+			return nil
+		}
+		return fmt.Errorf("rollout is in progress (CurrentRevision=%s UpdateRevision=%s)",
+			ss.Status.CurrentRevision, ss.Status.UpdateRevision)
+	}, logger)
 	if err != nil {
-		// This should not happen but if it does it's likely unrecoverable.
-		c.UI.Error(fmt.Sprintf("Error creating Consul API client: %s", err))
+		logger.Error(err.Error())
 		return 1
 	}
 
-	// Bootstrap ACLs.
-	bootstrapToken, err := c.bootstrapACLs(logger, consulClient)
+	// Check if we've already been bootstrapped.
+	bootTokenSecretName := fmt.Sprintf("%s-consul-bootstrap-acl-token", c.flagReleaseName)
+	bootstrapToken, err := c.getBootstrapToken(logger, bootTokenSecretName)
 	if err != nil {
-		c.UI.Error(err.Error())
+		logger.Error(fmt.Sprintf("Unexpected error looking for preexisting bootstrap Secret: %s", err))
 		return 1
 	}
 
-	// Override our original client with a new one that has the bootstrap token
-	// set.
-	consulClient, err = api.NewClient(&api.Config{
-		Address: firstServerAddr,
+	if bootstrapToken != "" {
+		logger.Info(fmt.Sprintf("ACLs already bootstrapped - retrieved bootstrap token from Secret %q", bootTokenSecretName))
+	} else {
+		logger.Info("No bootstrap token from previous installation found, continuing on to bootstrapping")
+		bootstrapToken, err = c.bootstrapServers(logger, bootTokenSecretName)
+		if err != nil {
+			logger.Error(err.Error())
+			return 1
+		}
+	}
+
+	// For all of the next operations we'll need a Consul client.
+	serverPods, err := c.getConsulServers(logger, 1)
+	if err != nil {
+		logger.Error(err.Error())
+		return 1
+	}
+	serverAddr := serverPods[0].Addr
+	consulClient, err := api.NewClient(&api.Config{
+		Address: serverAddr,
```

Comment:
I think if you allow my comment above of also checking for TLS `https` port and add attribute `Scheme` to podAddr struct, we could use:

```Scheme: serverPods[0].Scheme```

So this can be use for both `http` and `https` environments